### PR TITLE
add tests for UserSummary and related APIs

### PR DESCRIPTION
**Pull request by Jamiras**
API_GetUserPoints
API_GetUserProgress
API_GetUserRank
API_GetUserRecentlyPlayedGames
API_GetUserSummary

I originally started with the last one and the others were mostly covered by it in one way or another.

### CODE REVIEW

File: app_legacy/Helpers/database/player-achievement.php [hunk 1]
```diff
@@ -292,21 +289,25 @@ function getUsersRecentAwardedForGames(string $user, $gameIDsCSV, $numAchievemen
     // TODO: because of the "ORDER BY HardcoreAchieved", this query only returns non-hardcore
     //       unlocks if the user has more than $limit unlocks. Note that $limit appears to be
     //       default (5000) for all use cases except API_GetUserSummary
-    $query = "SELECT ach.ID, ach.GameID, gd.Title AS GameTitle, ach.Title, ach.Description, ach.Points, ach.BadgeName, (!ISNULL(aw.User)) AS IsAwarded, aw.Date AS DateAwarded, (aw.HardcoreMode) AS HardcoreAchieved
+    $query = "SELECT ach.ID, ach.GameID, gd.Title AS GameTitle, ach.Title, ach.Description, ach.Points, ach.BadgeName,
+                     aw.User AS IsAwarded, aw.Date AS DateAwarded, aw.HardcoreMode AS HardcoreAchieved
```
Comment (Jamiras) [inline app_legacy/Helpers/database/player-achievement.php:293]: sqllite didn't like the `!ISNULL` here, so I ended up fetching the column and doing the null check when fetching the rows.

File: app_legacy/Helpers/database/player-game.php [hunk 2]
```diff
@@ -119,39 +118,27 @@ function getUserProgress($user, $gameIDsCSV, &$dataOut): ?int
...
-
-    while ($data = mysqli_fetch_assoc($dbResult)) {
+    $dbResult = legacyDbFetchAll($query);
+    foreach ($dbResult as $data) {
         $dataOut[$data['GameID']]['NumPossibleAchievements'] = $data['AchCount'];
         $dataOut[$data['GameID']]['PossibleScore'] = $data['PointCount'];
-        $dataOut[$data['GameID']]['NumAchieved'] = 0;
-        $dataOut[$data['GameID']]['ScoreAchieved'] = 0;
-        $dataOut[$data['GameID']]['NumAchievedHardcore'] = 0;
-        $dataOut[$data['GameID']]['ScoreAchievedHardcore'] = 0;
```
Comment (Jamiras) [inline app_legacy/Helpers/database/player-game.php:137]: These are already initialized on lines 112-115

File: app_legacy/Helpers/database/player-rank.php [hunk 3]
```diff
@@ -136,34 +129,56 @@ function getTopUsersByScore($count, &$dataOut, $ofFriend = null): int
...
-function getUserRank(string $user, int $type = RankType::Hardcore): ?int
+function getUserRank(string $username, int $type = RankType::Hardcore): ?int
 {
-    return Cache::remember('user:' . $user . ':rank:' . ($type === RankType::Hardcore ? 'hardcore' : 'softcore'), Carbon::now()->addMinutes(15), function () use ($user, $type) {
-        sanitize_sql_inputs($user);
-
-        $joinCond = match ($type) {
-            default => "RIGHT JOIN UserAccounts AS ua2 ON ua.RAPoints < ua2.RAPoints AND NOT ua2.Untracked",
-            RankType::Softcore => "RIGHT JOIN UserAccounts AS ua2 ON ua.RASoftcorePoints < ua2.RASoftcorePoints AND NOT ua2.Untracked",
-            RankType::TruePoints => "RIGHT JOIN UserAccounts AS ua2 ON ua.TrueRAPoints < ua2.TrueRAPoints AND NOT ua2.Untracked",
```
Comment (Jamiras) [inline app_legacy/Helpers/database/player-rank.php:147]: sqllite didn't like the `RIGHT JOIN`, so I changed it to fetch a model and use properties from the model to query without a join.

File: app_legacy/Helpers/database/player-achievement.php (hunk 1 above)
Comment (luchaos) [reply app_legacy/Helpers/database/player-achievement.php:293]: Both mysql and sqlite accept `NOT NULL` as far as i know

**Review by luchaos — APPROVED** [review]
Awesome. I have another PR in preparation that further increases type strictness that has similar changes in it. I'll rebase that and have it ready soon.

**Pull request merged**